File: 0169-majority-element/0169-majority-element.py
class Solution:
    # def majorityElement(self, nums: List[int]) -> int:
    #   n = len(nums) // 2
    #   mp = {}
    #   for num in nums:
    #     mp[num] = mp.get(num, 0) + 1
    #     if (mp[num] > n):
    #       return num
    #   return -1
    def majorityElement(self, nums: List[int]) -> int:
      count = 0
      candidate = None

      for num in nums:
        if count == 0:
          candidate = num
        count += (1 if num == candidate else -1)
      return candidate

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
      int mid = nums.size()%2 == 0 ? nums.size()/2 - 1 : nums.size()/2;  
      sort(nums.begin(), nums.end());
      return nums[mid];
    }
};

// 4